### PR TITLE
Use CoordsXYZ for sub_98199C in RiverRapids.cpp

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

**Comment by frutiemax** [comment]
testpaint doing its own thing...

File: src/openrct2/paint/Paint.cpp [hunk 1]
```diff
@@ -986,6 +979,16 @@ paint_struct* sub_98199C(
...
+paint_struct* sub_98199C(
+    paint_session* session, uint32_t image_id, int8_t x_offset, int8_t y_offset, int16_t bound_box_length_x,
+    int16_t bound_box_length_y, int8_t bound_box_length_z, int16_t z_offset, int16_t bound_box_offset_x,
+    int16_t bound_box_offset_y, int16_t bound_box_offset_z)
+{
+    return sub_98199C(
+        session, image_id, { x_offset, y_offset, z_offset }, { bound_box_length_x, bound_box_length_y, bound_box_length_z },
+        { bound_box_offset_x, bound_box_offset_y, bound_box_offset_z });
+}
+
```
Comment (tupaschoal) [inline src/openrct2/paint/Paint.cpp:991]: Are there other users of this function? I don't get why this is being kept

File: src/openrct2/paint/Paint.cpp [hunk 2]
```diff
@@ -954,24 +954,17 @@ paint_struct* sub_98198C(
...
  */
 paint_struct* sub_98199C(
-    paint_session* session, uint32_t image_id, int8_t x_offset, int8_t y_offset, int16_t bound_box_length_x,
-    int16_t bound_box_length_y, int8_t bound_box_length_z, int16_t z_offset, int16_t bound_box_offset_x,
-    int16_t bound_box_offset_y, int16_t bound_box_offset_z)
+    paint_session* session, uint32_t image_id, const CoordsXYZ& offset, const CoordsXYZ& boundBoxLength,
+    const CoordsXYZ& boundBoxOffset)
 {
-    assert(static_cast<uint16_t>(bound_box_length_x) == static_cast<int16_t>(bound_box_length_x));
-    assert(static_cast<uint16_t>(bound_box_length_y) == static_cast<int16_t>(bound_box_length_y));
```
Comment (tupaschoal) [inline src/openrct2/paint/Paint.cpp:962]: Why removing the asserts?

Comment (Gymnasiast) [reply src/openrct2/paint/Paint.cpp:962]: Well, are they needed now we use signed 32-bit integers?

Comment (tupaschoal) [reply src/openrct2/paint/Paint.cpp:962]: `sub_98197C` below still take `int16_t`s so i'd say it's safer to keep them for now

File: src/openrct2/paint/Paint.cpp (hunk 1 above)
Comment (frutiemax) [reply src/openrct2/paint/Paint.cpp:991]: There are still a lot of remaining calls, too many for one PR.

**Comment by duncanspumpkin** [comment]
Looks fine to me.

**Pull request merged**